### PR TITLE
Fixed the double tap editor freeze issue and added test cases

### LINKED ISSUE #17757: [iOS] Editor autoscrolls its text view completely off the screen if you press 'return' too many times, has no constraint against scrolling the text view off screen. Objects lag in resizing around it.
### Description

A working Editor is essential for text input in a cross platform and mobile environment. Editor can be used as a field for users to type into such as in this forum box I am typing into now or a WhatsApp or SMS style text entry field. 

A typical such Editor function is described by:

```
            Editor editor = new();
            editor.AutoSize = EditorAutoSizeOption.TextChanges;
            editor.MaximumHeightRequest = 200;
```
In Windows and Android, Editor functions properly with such a setting. However in iOS it is quite broken, exhibiting at least three problems:

1. Editor in iOS will not scroll down automatically as you add new lines once it is at its max size (in Windows and Android, as you add more lines, the Editor automatically scrolls down to show them).

2. Editor in iOS does not respond to touch and drag to scroll manually (in Android, you can touch and drag to scroll and Windows use the mouse scroll wheel to scroll).

3. Parent element of Editor in iOS resizes with a one frame lag behind the Editor as the Editor expands, so as you add new lines and Editor resizes, Editor will momentarily overlap its parent object each time (in Android and Windows the resizing of parents are instant).

Android and Windows exhibit all the obvious desirable behaviors. 

That is: (i) A text field must automatically scroll down to always show the current line of text being written to, (ii) touching and dragging is the only way to manually scroll a text field in mobile and must work as expected, and (iii) resizing of the Editor should not introduce visual glitches and resize lags.

These problems only exist in iOS. Behavior of Editor in Android and Windows appears perfect.

### Steps to Reproduce

1. Create a new blank MAUI project in Visual Studio 2022 using .NET 7.0 by File > New.

2. Copy and paste the following to replace the default class in App.xaml.cs:
```
public partial class App : Application {
    public App() {
        InitializeComponent();

        MainPage = new ContentPage();

        VerticalStackLayout vert = new();
        (MainPage as ContentPage).Content = vert;
        
        Border border = new();
        border.StrokeThickness = 4;
        border.BackgroundColor = Colors.DarkBlue;
        border.Stroke = Colors.Red;
        border.Padding = 10;
        vert.Children.Add(border);

        Editor editor = new();
        editor.BackgroundColor = Colors.White;
        editor.AutoSize = EditorAutoSizeOption.TextChanges;
        editor.MaximumHeightRequest = 200;
        border.Content = editor;

        Label testLabel = new Label();
        testLabel.Text = "Editor iOS Bug Demonstration:\n- Editor does not autoscroll downward as you fill the Editor with text (new lines past max height go down off screen)\n- Editor does not scroll with click and drag once past maxHeight in size.\n- There is a frame lag after resizing Editor and before resizing the parent so the Editor goes 'out of bounds' of its parent momentarily each time it grows vertically.";
        vert.Children.Add(testLabel);

    }
}
```

3. Run the project in **Windows** and **Android** and you will observe the following **CORRECT BEHAVIOR**: 

- Editor will automatically scroll down to show the newly added text if you keep adding new lines once it is at max height.
- Editor scrolls properly from mouse scroll wheel function over the object in Windows, touch to drag in Android, and mouse scroll wheel over object if in Windows Android simulator.
- Editor as well as its parent Border will resize instantly in Windows and Android with no visible lag so the blue border around the Editor remains uniform at all times.

4. Run the project in **iOS** and observe by contrast the following **WRONG BEHAVIOR**: 

- Editor will not scroll down automatically as you add new lines once it is at max height.
- Editor does not respond at all to touch and drag to scroll manually.
- Border around Editor resizes with a one frame lag behind the Editor, so as you add new lines and the Editor expands, the Editor will momentarily overlap the blue background of Border at the bottom each time.

Expected behavior again is that Editor in iOS should show the same results as Windows and Android with: (i) automatic downward scrolling on new lines, (ii) manual scrolling on touch and drag (and scroll wheel if available from simulator), and (iii) instant resizing of its parents upon changing size.

### Link to public reproduction project repository

https://github.com/jonmdev/Editor-iOS-Bug

### Version with bug

7.0.92/8.0

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 16.7 on iPhone XR with Debug build (Hot Restart)

### Did you find any workaround?

None. 

### Relevant log output

_No response_

### LINKED ISSUE #25938: [iOS] Editor when entering a text which exceeds the maximum height resets the position
### Description

Create a Editor with MaximumHeight and AutoSize as TextChanges , when the text exceeds the maximum height ,each time when you enter the text , the frame gets resetted to the default position

### Steps to Reproduce

1.Create a MAUI project 
2. Add Editor with MaximumHeight and AutoSize as TextChanges 
`     
 <VerticalStackLayout>
            <Editor BackgroundColor="Yellow"
                    AutoSize="TextChanges"
                    MaximumHeightRequest="200"   />
        <Label Text="Editor iOS Bug Demonstration:
                    - Editor does maintains the position , each the frame get resetted to default)"/>
 </VerticalStackLayout>
`
3.Expected Behavior : Scroller should maintain the same position
4.Actual Behavior : Odd flicker due to reset , we need to scroll to see the text entered[TestMaui.zip](https://github.com/user-attachments/files/17811387/TestMaui.zip)

### Link to public reproduction project repository

_No response_

### Version with bug

9.0.10 SR1

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

```shell

```

### LINKED ISSUE #25940: iOS hangs when choosing to Select All text on an Editor when AutoSize = TextChanges
### Description

When adding an Editor to the AutoSize property set to TextChanges... adding a bunch of text and then tapping the editor to make use of the "Select All" option from UITextView freezes the app:

Image

### Steps to Reproduce

Craete the default new Maui app, and add an editor:

`<Editor HorizontalOptions="Fill" BackgroundColor="Yellow" AutoSize="TextChanges"/>`

When you type in a bunch of text, and the choose the "Select All" from the UITextView options, the app will freeze.

Full main page:
```
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="MauiApp2.MainPage">

        <VerticalStackLayout
            Padding="30,0"
            Spacing="25">
            <Image
                Source="dotnet_bot.png"
                HeightRequest="185"
                Aspect="AspectFit"
                SemanticProperties.Description="dot net bot in a race car number eight" />

            <Label
                Text="Hello, World!"
                Style="{StaticResource Headline}"
                SemanticProperties.HeadingLevel="Level1" />

            <Label
                Text="Welcome to &#10;.NET Multi-platform App UI"
                Style="{StaticResource SubHeadline}"
                SemanticProperties.HeadingLevel="Level2"
                SemanticProperties.Description="Welcome to dot net Multi platform App U I" />

            <Button
                x:Name="CounterBtn"
                Text="Click me" 
                SemanticProperties.Hint="Counts the number of times you click"
                Clicked="OnCounterClicked"
                HorizontalOptions="Fill" />

            <Editor HorizontalOptions="Fill" BackgroundColor="Yellow" AutoSize="TextChanges"/>
        </VerticalStackLayout>

</ContentPage>
```

### Link to public reproduction project repository

_No response_

### Version with bug

9.0.10 SR1

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 18.1

### Did you find any workaround?

Yes - see custom handler implementation in this [issue](https://github.com/dotnet/maui/issues/17757)

### Relevant log output

```shell

```

### LINKED ISSUE #25975: Double tapping Editor control locks app when deployed to iPad.
### Description

Double tapping the Editor locks the app up completely when deployed to iPad. This occurs for both net 8 and 9 and is reproducible using simulator and physical device. [iPhones do not appear to be affected by this.](https://learn.microsoft.com/en-us/answers/questions/2119016/double-tapping-editor-control-locks-app-when-deplo?comment=question#newest-question-comment) Entry control works well and does not lock up the app. Editor and Entry controls can be added to the demo Maui app as below to duplicate the issue.

### Steps to Reproduce

Add Editor and Entry controls to the demo maui app as below and run the app. Double tap on Entry control to verify it works well. Double tap on Editor control and lock your app up!

```
    <ScrollView>
        <VerticalStackLayout
            Padding="30,0"
            Spacing="25">
            <Image
                Source="dotnet_bot.png"
                HeightRequest="185"
                Aspect="AspectFit"
                SemanticProperties.Description="dot net bot in a hovercraft number nine" />

            <Label
                Text="Hello, World!"
                Style="{StaticResource Headline}"
                SemanticProperties.HeadingLevel="Level1" />

            <Label
                Text="Welcome to &#10;.NET Multi-platform App UI"
                Style="{StaticResource SubHeadline}"
                SemanticProperties.HeadingLevel="Level2"
                SemanticProperties.Description="Welcome to dot net Multi platform App U I" />

            <Button
                x:Name="CounterBtn"
                Text="Click me" 
                SemanticProperties.Hint="Counts the number of times you click"
                Clicked="OnCounterClicked"
                HorizontalOptions="Fill" />

            <Editor Text="Hello" FontSize="24" VerticalOptions="Center" HorizontalTextAlignment="Center" HorizontalOptions="FillAndExpand" AutoSize="TextChanges"/>

            <Entry Text="Hello" FontSize="24" VerticalOptions="Center" HorizontalTextAlignment="Center" HorizontalOptions="FillAndExpand" />

        </VerticalStackLayout>
    </ScrollView>
,,,

### Link to public reproduction project repository

_No response_

### Version with bug

9.0.10 SR1

### Is this a regression from previous behavior?

Not sure, I know its present in net8 and net9

### Last version that worked well

unknown, i only started maui at net8

### Affected platforms

iOS, specifically iPadOS which is slightly different than iPhoneOS

### Affected platform versions

iPadOS

### Did you find any workaround?

No work around yet

### Relevant log output

```shell

```

### PR DESCRIPTION
**Pull request by devanathan-vaithiyanathan**
### Issue Details
Double-tapping the editor causes the app to freeze or become unresponsive.

### Root Cause
When you double-tap the editor and AutoSize is set to "TextChanges," the editor continuously resizes to fit the content, leading to the app freezing or hanging.
 
### Description of Change
* Avoided calling PlatformView.SizeToFit() directly within the GetDesiredSize() method. Instead, a more controlled approach is used to measure the size.

### Issues Fixed

Fixes #25975 
Fixes https://github.com/dotnet/maui/issues/25940
Fixes https://github.com/dotnet/maui/issues/25938
Fixes https://github.com/dotnet/maui/issues/17757

### Tested the behaviour in the following platforms
 
- [x] Android
- [x] Windows
- [x] iOS
- [x] Mac

### Output Screenshot
| Before  | After  |
|---------|--------|
|  <video src="https://github.com/user-attachments/assets/1d772981-47cd-4ff0-9f27-2fea4960f97e" width="320" height="640" controls></video>   |   <video src="https://github.com/user-attachments/assets/2382d890-6297-4940-918b-911d17f23b86" width="320" height="640" controls></video>   |

### CODE REVIEW

File: src/Core/src/Handlers/Editor/EditorHandler.iOS.cs [hunk 1]
```diff
@@ -61,16 +61,16 @@ public override Size GetDesiredSize(double widthConstraint, double heightConstra
 				// get an exception; it doesn't know what do to with it. So instead we'll size
 				// it to fit its current contents and use those values to replace infinite constraints
 
-				PlatformView.SizeToFit();
+				var sizeThatFits = PlatformView.SizeThatFits(new CGSize(widthConstraint, heightConstraint));
```
Comment (jsuarezruiz) [inline src/Core/src/Handlers/Editor/EditorHandler.iOS.cs:64]: Exactly the same logic is applied in the SearchBar, happens the same problem? does it require the same changes?

https://github.com/dotnet/maui/blob/489657c5458691f3a4c360d84b3a53a03b61b03f/src/Core/src/Handlers/SearchBar/SearchBarHandler.iOS.cs#L43

If need the same changes, could you add a test covering the fix in the SearchBar?

**Comment by Shalini-Ashokan** [comment]
@jsuarezruiz, Since the issue arises when AutoSize is set in the editor, and AutoSize is not supported in the search bar, the search bar does not exhibit the freezing behavior and functions as expected.

**Review by PureWeen — APPROVED** [review]

**Pull request merged**